app/javascript/components/explore/ExploreDisplayTabs.jsx

```diff
@@ -36,19 +36,29 @@ import Tooltip from 'react-bootstrap/lib/Tooltip'
 import DifferentialExpressionModal from '~/components/explore/DifferentialExpressionModal'
 import PlotTabs from './PlotTabs'
 
-/** Determine if currently selected cluster has differential expression outputs available */
-function getClusterHasDe(exploreInfo, exploreParams) {
-  const flags = getFeatureFlagsWithDefaults()
-  if (!flags?.differential_expression_frontend || !exploreInfo) {return false}
-  let clusterHasDe = false
+/** Get the selected clustering and annotation, or their defaults */
+function getSelectedClusterAndAnnot(exploreInfo, exploreParams) {
   const annotList = exploreInfo.annotationList
   let selectedCluster
+  let selectedAnnot
   if (exploreParams?.cluster) {
     selectedCluster = exploreParams.cluster
+    selectedAnnot = exploreParams.annotation
   } else {
     selectedCluster = annotList.default_cluster
+    selectedAnnot = annotList.default_annotation
   }
 
+  return [selectedCluster, selectedAnnot]
+}
+
+/** Determine if currently selected cluster has differential expression outputs available */
+function getClusterHasDe(exploreInfo, exploreParams) {
+  const flags = getFeatureFlagsWithDefaults()
+  if (!flags?.differential_expression_frontend || !exploreInfo) {return false}
```

Comment:
Thinking out loud - I think we're fairly invested in differential expression visualizations.  Should we retire this feature flag?

Reply:
I had the same thought, then found some prior discussion that we intend to retain the flag so we can easily exempt public studies from displaying DE if the owner requests so.  A new [code comment](https://github.com/broadinstitute/single_cell_portal_core/pull/1806/files#diff-d56fa6c127e17ffa714e80c992b809c04b0dda9a45a93425adca8a56f0d607c2R203) notes this too.

Reply:
Oh that's a great point - I agree with that.